### ISSUE TITLE
[Issue]: CleanupCollectionAndPlaylistPathsTask removing series / directories

### ISSUE THREAD

**Issue by jmqm** [jellyfin/jellyfin#11289]
### Please describe your bug

When running task `CleanupCollectionAndPlaylistPathsTask`, it may remove items in collections (and maybe playlists) that contain foreign language, one of them being Korean. I don't know other languages that may be impacted.

### Reproduction Steps

1. Add an item to a library with foreign language (e.g. `The Glory (더 글로리) (2022)`)
2. Add the item to a collection
3. Run task `CleanupCollectionAndPlaylistPathsTask`

### Jellyfin Version

Unstable (master branch)

### Environment

```markdown
- Virtualization: Docker (jellyfin/jellyfin:unstable 2024/04/01)
- Browser: Firefox 124
- Storage: Docker local volume
```

### Jellyfin logs

```shell
[2024-04-03 13:42:04.722 -05:00] [DBG] [25] Emby.Server.Implementations.ScheduledTasks.TaskManager: Executing "Clean up collections and playlists"
[2024-04-03 13:42:04.724 -05:00] [DBG] [25] Emby.Server.Implementations.ScheduledTasks.Tasks.CleanupCollectionAndPlaylistPathsTask: Found 3 boxsets
[...]
[2024-04-03 13:42:04.730 -05:00] [DBG] [25] Emby.Server.Implementations.ScheduledTasks.Tasks.CleanupCollectionAndPlaylistPathsTask: Checking boxset "Korean"
[...]
[2024-04-03 13:42:04.731 -05:00] [INF] [25] Emby.Server.Implementations.ScheduledTasks.Tasks.CleanupCollectionAndPlaylistPathsTask: Item in "Korean" cannot be found at "/media/TV Series/Weak Hero (약한영웅) (2022)"
[2024-04-03 13:42:04.731 -05:00] [INF] [25] Emby.Server.Implementations.ScheduledTasks.Tasks.CleanupCollectionAndPlaylistPathsTask: Item in "Korean" cannot be found at "/media/TV Series/My Name (마이 네임) (2021)"
[2024-04-03 13:42:04.731 -05:00] [INF] [25] Emby.Server.Implementations.ScheduledTasks.Tasks.CleanupCollectionAndPlaylistPathsTask: Item in "Korean" cannot be found at "/media/TV Series/The Glory (더 글로리) (2022)"
[2024-04-03 13:42:04.731 -05:00] [DBG] [25] Emby.Server.Implementations.ScheduledTasks.Tasks.CleanupCollectionAndPlaylistPathsTask: Updating "Korean"
[2024-04-03 13:42:04.731 -05:00] [DBG] [25] MediaBrowser.Providers.Manager.ProviderManager: Saving "/data/data/collections/Korean" to "Emby Xml"
[...]
[2024-04-03 13:42:04.734 -05:00] [INF] [25] Emby.Server.Implementations.ScheduledTasks.TaskManager: "Clean up collections and playlists" Completed after 0 minute(s) and 0 seconds
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jmqm** [comment]
Seems like it only affects series, movies were not being removed.

**Comment by jmqm** [comment]
<details>
  <summary>
    Original
  </summary>

```
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<Item>
  <ContentRating>TV-MA</ContentRating>
  <Added>05/15/2023 18:34:38</Added>
  <LockData>true</LockData>
  <LocalTitle>Korean</LocalTitle>
  <PremiereDate>2016-05-25</PremiereDate>
  <DisplayOrder>PremiereDate</DisplayOrder>
  <ProductionYear>2016</ProductionYear>
  <Genres>
    <Genre>Action</Genre>
    <Genre>Crime</Genre>
    <Genre>Drama</Genre>
    <Genre>Adventure</Genre>
    <Genre>Thriller</Genre>
    <Genre>Mystery</Genre>
    <Genre>Suspense</Genre>
    <Genre>Comedy</Genre>
    <Genre>Romance</Genre>
    <Genre>Science Fiction</Genre>
    <Genre>History</Genre>
    <Genre>War</Genre>
  </Genres>
  <Studios>
    <Studio>Netflix</Studio>
    <Studio>TVING</Studio>
    <Studio>Disney+</Studio>
    <Studio>tvN</Studio>
    <Studio>Moho Film</Studio>
    <Studio>Yong Film</Studio>
    <Studio>CJ Entertainment</Studio>
    <Studio>Lotte Entertainment</Studio>
    <Studio>Big Stone Pictures</Studio>
    <Studio>Next Entertainment World</Studio>
    <Studio>M Pictures</Studio>
    <Studio>Barunson E&amp;A</Studio>
  </Studios>
</Item>
```
</details>

<details>
  <summary>
    After I added a Korean series
  </summary>

```
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<Item>
  <ContentRating>TV-MA</ContentRating>
  <Added>05/15/2023 18:34:38</Added>
  <LockData>true</LockData>
  <LocalTitle>Korean</LocalTitle>
  <PremiereDate>2016-05-25</PremiereDate>
  <DisplayOrder>PremiereDate</DisplayOrder>
  <ProductionYear>2016</ProductionYear>
  <Genres>
    <Genre>Action</Genre>
    <Genre>Crime</Genre>
    <Genre>Drama</Genre>
    <Genre>Adventure</Genre>
    <Genre>Thriller</Genre>
    <Genre>Mystery</Genre>
    <Genre>Suspense</Genre>
    <Genre>Comedy</Genre>
    <Genre>Romance</Genre>
    <Genre>Science Fiction</Genre>
    <Genre>History</Genre>
    <Genre>War</Genre>
  </Genres>
  <Studios>
    <Studio>Netflix</Studio>
    <Studio>TVING</Studio>
    <Studio>Disney+</Studio>
    <Studio>tvN</Studio>
    <Studio>Moho Film</Studio>
    <Studio>Yong Film</Studio>
    <Studio>CJ Entertainment</Studio>
    <Studio>Lotte Entertainment</Studio>
    <Studio>Big Stone Pictures</Studio>
    <Studio>Next Entertainment World</Studio>
    <Studio>M Pictures</Studio>
    <Studio>Barunson E&amp;A</Studio>
  </Studios>
  <CollectionItems>
    <CollectionItem>
      <Path>/media/TV Series/The Glory (더 글로리) (2022)</Path>
    </CollectionItem>
  </CollectionItems>
</Item>
```
</details>

<details>
  <summary>
    After running task CleanupCollectionAndPlaylistPathsTask
  </summary>

```
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<Item>
  <ContentRating>TV-MA</ContentRating>
  <Added>05/15/2023 13:34:38</Added>
  <LockData>true</LockData>
  <LocalTitle>Korean</LocalTitle>
  <PremiereDate>2016-05-24</PremiereDate>
  <DisplayOrder>PremiereDate</DisplayOrder>
  <ProductionYear>2016</ProductionYear>
  <Genres>
    <Genre>Action</Genre>
    <Genre>Crime</Genre>
    <Genre>Drama</Genre>
    <Genre>Adventure</Genre>
    <Genre>Thriller</Genre>
    <Genre>Mystery</Genre>
    <Genre>Suspense</Genre>
    <Genre>Comedy</Genre>
    <Genre>Romance</Genre>
    <Genre>Science Fiction</Genre>
    <Genre>History</Genre>
    <Genre>War</Genre>
  </Genres>
  <Studios>
    <Studio>Netflix</Studio>
    <Studio>TVING</Studio>
    <Studio>Disney+</Studio>
    <Studio>tvN</Studio>
    <Studio>Moho Film</Studio>
    <Studio>Yong Film</Studio>
    <Studio>CJ Entertainment</Studio>
    <Studio>Lotte Entertainment</Studio>
    <Studio>Big Stone Pictures</Studio>
    <Studio>Next Entertainment World</Studio>
    <Studio>M Pictures</Studio>
    <Studio>Barunson E&amp;A</Studio>
  </Studios>
</Item>
```
</details>

I did some testing, adding an English series, episode and movie in a test collection. When running task `CleanupCollectionAndPlaylistPathsTask`, the series was removed but the others were unaffected. The problem might be series in general, not anything to do with foreign languages.

Here's a test of adding a variety of media:

<details>
  <summary>
    Added series, episodes and movies to a collection
  </summary>

```
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<Item>
  <ContentRating>CA-PG</ContentRating>
  <Added>04/05/2024 19:13:39</Added>
  <LockData>true</LockData>
  <LocalTitle>Test Collection</LocalTitle>
  <DisplayOrder>PremiereDate</DisplayOrder>
  <CollectionItems>
    <CollectionItem>
      <Path>/media/Anime/BOCCHI THE ROCK! (ぼっち・ざ・ろっく!) (2022)</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Anime/BOCCHI THE ROCK! (ぼっち・ざ・ろっく!) (2022)/Season 01/Bocchi the Rock! (2022) - S01E01 - Lonely Rolling Bocchi.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/TV Series/The Glory (더 글로리) (2022)</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/TV Series/The Glory (더 글로리) (2022)/Season 01/The Glory (2022) - S01E01 - Episode 1.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Anime/Akira (AKIRA) (1988)/Akira (AKIRA) (1988).mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/The Batman (2022)/The Batman (2022) - 2160p (4K), HDR.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/TV Series/Gen V (2023)</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/TV Series/Gen V (2023)/Season 01/Gen V (2023) - S01E01 - God U.mkv</Path>
    </CollectionItem>
  </CollectionItems>
</Item>
```
</details>

<details>
  <summary>
    After running task CleanupCollectionAndPlaylistPathsTask
  </summary>

```
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<Item>
  <ContentRating>CA-PG</ContentRating>
  <Added>04/05/2024 14:13:39</Added>
  <LockData>true</LockData>
  <LocalTitle>Test Collection</LocalTitle>
  <DisplayOrder>PremiereDate</DisplayOrder>
  <CollectionItems>
    <CollectionItem>
      <Path>/media/Anime/BOCCHI THE ROCK! (ぼっち・ざ・ろっく!) (2022)/Season 01/Bocchi the Rock! (2022) - S01E01 - Lonely Rolling Bocchi.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/TV Series/The Glory (더 글로리) (2022)/Season 01/The Glory (2022) - S01E01 - Episode 1.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Anime/Akira (AKIRA) (1988)/Akira (AKIRA) (1988).mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/Movies/The Batman (2022)/The Batman (2022) - 2160p (4K), HDR.mkv</Path>
    </CollectionItem>
    <CollectionItem>
      <Path>/media/TV Series/Gen V (2023)/Season 01/Gen V (2023) - S01E01 - God U.mkv</Path>
    </CollectionItem>
  </CollectionItems>
</Item>
```
</details>

**Comment by JPVenson** [comment]
Ahh god point. Thank you a lot, i figured out whats the issue here and will soon create a fix for that.

TL;DR: i forgot to take directories into account when writing the cleanup task. Wups :-D
